aspnetcore/security/authentication/certauth.md

```diff
@@ -0,0 +1,243 @@
+monikerRange: '>= aspnetcore-3.0'
+ms.author: barry.dorrans
+ms.date: 06/10/2019
+uid: security/authentication/certauth
+---
+# Overview
+
+`Microsoft.AspNetCore.Authentication.Certificate` sort of contains an implementation of [Certificate Authentication](https://tools.ietf.org/html/rfc5246#section-7.4.4) for ASP.NET Core. 
+Certificate authentication happens at the TLS level, long before it ever gets to ASP.NET Core, so, more accurately this is an authentication handler
+that validates the certificate and then gives you an event where you can resolve that certificate to a ClaimsPrincipal. 
+
+You **must** [configure your host](#hostConfiguration) for certificate authentication, be it IIS, Kestrel, Azure Web Applications or whatever else you're using.
+
+## Getting started
+
+First acquire an HTTPS certificate, apply it and then [configure your host](#hostConfiguration) to require certificates.
+
+In your web application add a reference to the package, then in the `ConfigureServices` method in `startup.cs` call
+`app.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme).UseCertificateAuthentication(...);` with your options, 
+providing a delegate for `OnValidateCertificate` to validate the client certificate sent with requests and turn that information 
+into an `ClaimsPrincipal`, set it on the `context.Principal` property and call `context.Success()`.
+
+If you change your scheme name in the options for the authentication handler you need to change the scheme name in 
+`AddAuthentication()` to ensure it's used on every request which ends in an endpoint that requires authorization.
+
+If authentication fails this handler will return a `403 (Forbidden)` response rather a `401 (Unauthorized)` as you
+might expect - this is because the authentication should happen during the initial TLS connection - by the time it 
+reaches the handler it's too late, and there's no way to actually upgrade the connection from an anonymous connection 
+to one with a certificate.
+
+You must also add `app.UseAuthentication();` in the `Configure` method, otherwise nothing will ever get called.
+
+For example;
+
+```c#
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme)
+            .AddCertificate();
+    // All the other service configuration.
+}
+
+public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+{
+    app.UseAuthentication();
+
+    // All the other app configuration.
+}
+```
+
+In the sample above you can see the default way to add certificate authentication. The handler will construct a user principal using the common certificate properties for you.
+
+## Configuring Certificate Validation
+
+The `CertificateAuthenticationOptions` handler has some built in validations that are the minimium validations you should perform on 
+a certificate. Each of these settings are turned on by default.
+
+### ValidateCertificateChain
```

Comment:
@blowdart I don't see any setting like this in the Options class unlike the other ones, is this gone?

Reply:
It moved to  CertificateTypes AllowedCertificateTypes { get; set; } = CertificateTypes.Chained, because if it's chained you should be trusting the chain

Reply:
Ok will update